### ISSUE TITLE
Wire MultiTaskLoss through the training loop

### ISSUE THREAD

**Issue by yusufizzetmurat** [yusufizzetmurat/fed-pulse#273]
PR #272 landed the `MultiTaskHead` + `MultiTaskLoss` modules and the per-axis target extraction on the loader. What is missing is the DataLoader plumbing: the multi-task targets + masks need to ride alongside `(x, y, text_emb, text_missing)` through `_build_partition_tensors` → `_unpack_batch` → the training step, and the loss has to be constructed per-fold with class weights fitted per axis (stance / certainty / topic).

### Files to touch

- `backend/app/training/loop.py` — extend `_build_partition_tensors` to also call `_build_multi_task_target_tensors` and return the 8 tensors alongside the existing tuple. Update `_unpack_batch` to handle the wider tuple. In `train_model`, swap the single `CrossEntropyLoss` for `MultiTaskLoss` when `output_mode == "classification"`, fitting per-axis class weights via `fit_class_weights` on the masked rows for stance / certainty / topic.
- `backend/app/train_forecaster.py` — surface `--multi-task-loss` flag so the per-axis loss is opt-in until the joint training has been validated; default off so the headline stance F1 stays stable.
- `backend/app/training/checkpoint.py` — persist per-axis class weights + lambdas onto the checkpoint payload so resume reads back the same loss config.

### Verification

- `docker compose run --rm backend pytest tests/unit/test_multi_task_head_shape.py tests/unit/test_multi_task_loss_masking.py tests/unit/test_loader_emits_per_axis_targets.py -q`
- One small GPU run (Transformer, seed 97, 4 folds) with `--multi-task-loss on` confirming stance macro-F1 stays within ±0.005 of the Path B 0.5225 baseline.
- Wiki §6.8 appendix table with per-axis macro-F1 + bootstrap CI.

### Out of scope

A text-only multi-axis classifier (FinBERT + multi-task head) lands as a separate issue.
